### PR TITLE
Fixing NPE on ReplicatedMap TTL eviction

### LINKED ISSUE #6549: [TEST-FAILURE] ReplicatedMapTest.testEvictionObject
```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds , count left: 1
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:748)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:741)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:733)
	at com.hazelcast.replicatedmap.ReplicatedMapTest.testEviction(ReplicatedMapTest.java:967)
	at com.hazelcast.replicatedmap.ReplicatedMapTest.testEvictionObject(ReplicatedMapTest.java:934)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-IbmJDK1.7/com.hazelcast$hazelcast/742/testReport/junit/com.hazelcast.replicatedmap/ReplicatedMapTest/testEvictionObject/

### LINKED ISSUE #6776: 3.6-RC1  NPE replicatedmap.impl.operation.PutOperation.run
FAIL http://54.87.52.100/~jenkins/workspace/Hazelcast-Simulator-nightly/3.6-RC1/2015_11_18-16_42_49

full logs of run 
http://54.87.52.100/~jenkins/workspace/Hazelcast-Simulator-nightly/3.6-RC1/2015_11_18-16_42_49/

Test
com.hazelcast.simulator.tests.replicatedmap.ReplicatedMapTimeToLiveTest

```
java.lang.NullPointerException
	at com.hazelcast.replicatedmap.impl.record.AbstractReplicatedRecordStore.put(AbstractReplicatedRecordStore.java:157)
	at com.hazelcast.replicatedmap.impl.operation.PutOperation.run(PutOperation.java:61)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:168)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThread(ClassicOperationExecutor.java:385)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThreadIfPossible(ClassicOperationExecutor.java:345)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:259)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:243)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invokeInternal(Invocation.java:210)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:180)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:46)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:65)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:120)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:100)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:125)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processPartitionSpecificRunnable(OperationThread.java:159)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:142)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:124)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:99)
	at ------ End remote and begin local stack-trace ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveResponse(ClientInvocationFuture.java:133)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:100)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:75)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:38)
	at com.hazelcast.client.spi.ClientProxy.invokeOnPartition(ClientProxy.java:126)
	at com.hazelcast.client.spi.ClientProxy.invoke(ClientProxy.java:120)
	at com.hazelcast.client.proxy.ClientReplicatedMapProxy.put(ClientReplicatedMapProxy.java:120)
	at com.hazelcast.simulator.tests.replicatedmap.ReplicatedMapTimeToLiveTest$Worker.timeStep(ReplicatedMapTimeToLiveTest.java:111)
	at com.hazelcast.simulator.tests.replicatedmap.ReplicatedMapTimeToLiveTest$Worker.timeStep(ReplicatedMapTimeToLiveTest.java:94)
	at com.hazelcast.simulator.worker.tasks.AbstractWorker.run(AbstractWorker.java:77)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.simulator.utils.ThreadSpawner$ReportExceptionThread.run(ThreadSpawner.java:183)
```
http://54.87.52.100/~jenkins/workspace/Hazelcast-Simulator-nightly/3.6-RC1/2015_11_18-16_42_49/failures-2015-11-18__16_43_04.txt

### PR DESCRIPTION
**Pull request by eminn**
Fixed NPE and race condition between put and eviction operations by making evictions on partition threads.

Fixes #6776
Fixes #6549

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTtlTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+//                        ReplicatedRecordStore store = getStore(map, key);
+//                        ReplicatedRecord<Object, Object> replicatedRecord = getReplicatedRecord(map, key);
+//                        System.out.println("replicatedRecord = " + replicatedRecord);
+//                    }
+//                }
+//                sleepAtLeastSeconds(1);
+//            }
+//
+//        }
+
```
Comment (eminn) [inline hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTtlTest.java]: These will be removed...

**Comment by eminn** [comment]
run-lab-run

**Comment by eminn** [comment]
run-lab-run

**Comment by hasancelik** [comment]
run-lab-run

**Comment by gurbuzali** [comment]
:+1:

**Comment by metanet** [comment]
looks good :+1:

**Pull request merged**